benchmarks/benchmark_low_bit_adam.py

```diff
@@ -248,15 +249,26 @@ def evaluate_model(model, args):
         model.train()
         pbar = tqdm(dloader, dynamic_ncols=True, desc=f"Epoch {epoch_idx + 1}/{args.n_epochs}")
 
-        with torch.profiler.profile() if args.profile else nullcontext() as prof:
+        if args.profile:
+            activities = [torch.profiler.ProfilerActivity.CPU]
+            if args.device == "cuda":
+                activities.append(torch.profiler.ProfilerActivity.CUDA)
+            elif args.device == "xpu":
+                activities.append(torch.profiler.ProfilerActivity.XPU)
+
+            prof = torch.profiler.profile(activities=activities)
+        else:
```

Comment:
Can you check if on XPU PyTorch, PyTorch can auto-detect `ProfilerActivity.XPU`? https://github.com/pytorch/pytorch/blob/a8ab6b0938ed1788b0cad8982bcafaeaa5dea77d/torch/profiler/profiler.py#L61
Then we don't need to do that manually ourselves here.

Reply:
Checked it works w/ `torch.profiler.supported_activities()` to auto-detect `ProfilerActivity`. THX for ur review comment.

Reply:
Since that works, I think we can just call `torch.profiler.profile()`. It will call `torch.profiler.supported_activities()` internally if `activities` is empty.

https://github.com/pytorch/pytorch/blob/a87925cc7e3a617542204e5f7f8e8ca766caaefe/torch/profiler/profiler.py#L669